app/models/mixins/container_resource_parent_mixin.rb

```diff
@@ -0,0 +1,109 @@
+# ContainerResourceParentMixin provides capabilities to container
+# resources that provide hierarchical management to other container
+# resources.
+module ContainerResourceParentMixin
+  extend ActiveSupport::Concern
+  require 'json'
+  def get_resource_by_name(name, kind, api_version = 'v1')
+    resources = get_resources(kind, api_version)
+    resources.each do |resource|
+      if resource[:metadata][:name].eql? name
+        return resource
+      end
+    end
+    nil
+  end
+
+  def get_resources(kind, api_version = 'v1')
+    resources = []
+    method_name = "get_#{kind.underscore.pluralize}"
+    response, error_code, msg = send_method(method_name, [:namespace => name], api_version)
+    if error_code != 200
+      raise MiqException::MiqProvisionError, "Unexpected Exception while getting objects with method name #{method_name}, HTTP response code #{error_code}, and message #{msg}."
+    else
+      response.each do |object|
+        resources.push(object.to_h)
+      end
+    end
+    resources
+    # begin
+    #  objects = ext_management_system.connect_client(api_version, method_name).send(method_name, :namespace => name)
+    #  objects.each do |object|
+    #    resources.push(object.to_h)
+    #  end
+    # rescue KubeException => e
+    #  raise MiqException::MiqProvisionError, "Unexpected Exception while getting objects of kind #{kind} by method #{method_name}: #{e}"
+    # end
+  end
+
+  def patch_resource(obj)
+    obj = obj.symbolize_keys
+    obj[:metadata][:namespace] = name
+    obj_name = obj[:metadata][:name]
+    method_name = "patch_#{obj[:kind].underscore}"
+    # params = [obj_name, obj, name]
+    response, error_code, msg = send_method(method_name, [obj_name, obj, name], obj[:apiVersion])
+    if error_code != 200
+      raise MiqException::MiqProvisionError, "Unexpected Exception while patching object with method name #{method_name}, HTTP response code #{error_code}, and message #{msg}."
+    end
+    response
+    # begin
+    #  ext_management_system.connect_client(obj[:apiVersion], method_name).send(method_name.to_sym, *params)
+    # rescue KubeException => e
+    #   raise MiqException::MiqProvisionError, "Unexpected Exception while patching object: #{e}"
+    # end
+  end
+
+  def create_resource(obj)
+    obj = obj.symbolize_keys
+    obj[:metadata][:namespace] = name
+    method_name = "create_#{obj[:kind].underscore}"
```

Comment:
I'm adding a mapping between openshift/k8s names to miq names in https://github.com/ManageIQ/manageiq/pull/13619  (not merged yet).